src/Context/cartContext.jsx

```diff
@@ -0,0 +1,133 @@
+import { createContext, useContext, useReducer } from "react";
+import { cartReducer } from "../Reducer/cartReducer";
+import {useAuth} from "../Context/authContext";
+import { useEffect } from "react";
+import axios from "axios";
+
+
+
+const CartContext = createContext();
+
+const CartProvider = ({children}) => {
+
+   
+   
+    const [cartState, cartDispatch] = useReducer(cartReducer, {
+        cart: []
+    })
+    const {token, user} = useAuth();
+
+    useEffect(() => {
+        {
+            token && 
+            (async() => {
+                try {
+                    const {data: {cart} } = await axios("api/user/cart", {
+                        headers: {
+                            authorization: token,
+                        }                   
+                    })
+                    console.log(cart)
+                    cartDispatch({type: "GET_CARTITEM_FROM_LOCAL_STORAGE", payload: cart})
+                }catch(error){
```

Comment:
Cart Data is coming from backend not from LocalStorage , keep the naming conventions correct